DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,322 @@
+        long payload = (long)day;
+
+        //Process Year Value
+        if (year >= LEAP_YEAR_START){
+            while (y < year){
+               if (isLeapYear(y)){
+                    payload += (long)DAYS_IN_LEAP_YEAR;
+                } else {
+                    payload += (long)DAYS_IN_REG_YEAR;
+                } 
+                y++;
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid Year entered");
+        }
+
+        // //Process Month Value
+        for (int i = 0; i < month - 1; i++){
+            if(isLeapYear(year) && Date.Month.values()[i] == Month.FEB){
+                payload += Date.Month.values()[i].days + 1;
+            } else {
+                payload += Date.Month.values()[i].days;
+            }
+        }
+
+        return payload;
+    }
+
+    public static boolean isLeapYear(int year){
+        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+    }
+
+    public enum Month {
+        JAN("Jan", "January", 31),
+        FEB("Feb", "Febuary", 28),
+        MAR("Mar", "March", 31),
+        APR("Apr", "April", 30),
+        MAY("May", "May", 31),
+        JUN("Jun", "June", 30),
+        JUL("Jul", "July", 31),
+        AUG("Aug", "August", 31),
+        SEP("Sep", "September", 30),
+        OCT("Oct", "October", 31),
+        NOV("Nov", "November", 30),
+        DEC("Dec", "December", 31),
+        ;
+
+        public final String shortName;
+        public final String longName;
+        public final int days;
+
+        private Month(final String shortName, final String longName, final int days){
+            this.shortName = shortName;
+            this.longName = longName;
+            this.days = days;
+        }
+
+        //CAL~Already exists under enum class[compareTO] made class to see how we would implement if required
+        // public int diff(Month input){
+        //     return this.ordinal() - input.ordinal();
+
+        //     // if (this.ordinal() == input.ordinal()) {
+        //     //     return 0;
+        //     // } else if (this.ordinal() > input.ordinal()){
+        //     //     return 1;
+        //     // } else {
+        //     //     return -1;
+        //     // }
+        // }
+    }
+
+    private int dayspan;
+    private int year;
+    private Month month;
+    private int day;
+
+// Constructors
+    private Date(){
+        this.dayspan = 0;
+    }
+
+    public Date(int year, Month month, int day){
+        setYear(year);
+        setDay(day);
+        setMonth(month);
+    }
+
+    public Date(Month month){
+        setMonth(month);
+    }
+
+    public Date(int input){
+        setDayspan(input);
+    }
+// End Constructors
+
+// Methods
+    public Date subtract(int years, int months, int days){
+        int y = this.year;
+        int m = this.month.ordinal() + 1;
+        int d = this.day;
+
+        if (years >= 0 && months >= 0 && days >= 0) {
+            y -= years;
```

Comment:
What happens on this case?
```java
Date dt = new Date(1900, Date.Month.FEB, 1);
dt.subtract(3000,1,1);
```

Reply:
I've rewritten this whole method and would like to discuss it's design and function during our next meeting.